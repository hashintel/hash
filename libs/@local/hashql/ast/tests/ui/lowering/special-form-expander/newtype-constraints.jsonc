//@ run: pass
//@ description: Constraints should be converted to an AST node
["::kernel::special_form::newtype", "Foo<T, U: Bar>", "Integer", "_"]
