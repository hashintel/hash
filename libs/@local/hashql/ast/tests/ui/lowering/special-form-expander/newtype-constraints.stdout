Expr#4294967040@28
  ExprKind (NewType)
    NewTypeExpr#4294967040@28 (name: Foo)
      GenericConstraint#4294967040@14 (name: T)
      GenericConstraint#4294967040@19 (name: U)
        Type#4294967040@18
          TypeKind (Path)
            Path#4294967040@18 (rooted: false)
              PathSegment#4294967040@17 (name: Bar)
      Type#4294967040@25
        TypeKind (Path)
          Path#4294967040@25 (rooted: false)
            PathSegment#4294967040@24 (name: Integer)
      Expr#4294967040@27
        ExprKind (Underscore)
